configs/projects/puppet-agent.rb

```diff
@@ -253,4 +253,6 @@
   proj.directory proj.logdir unless platform.is_windows?
   proj.directory proj.piddir unless platform.is_windows?
 
+  proj.timeout 7200 if platform.is_windows?
```

Comment:
You won't monkey patch Vanagon from inside a component  to make the timeout exponentially increase on every failure. You're scared.